text/0059-geo-search.md

```diff
@@ -0,0 +1,232 @@
+```
+
+**`_geo` column definition**
+
+- Name: `_geo`
+- Type: String
+- Format: `"lat:float,lng:float"`
+- Not required
+
+#### POST Add or replace documents `/indexes/{indexUid}/documents`
+
+##### Request body
+```
+[
+    {
+        "id": 1,
+        "label": "F40",
+        "brand": "Ferrari",
+        "_geo": {
+            "lat": 48.862725,
+            "lng": 2.287592
+        }
+    }
+]
+```
+
+##### 202 Accepted - Response body
+
+```
+{
+    "updateId": 1
+}
+```
+
+#### PUT Add or replace documents `/indexes/{indexUid}/documents`
+
+##### Request body
+```
+[
+    {
+        "id": 1,
+        "brand": "F40 LM",
+        "brand": "Ferrari",
+        "_geo": {
+            "lat": 48.862725,
+            "lng": 2.287592
+        }
+    }
+]
+```
+
+##### 202 Accepted - Response body
+
+```
+{
+    "updateId": 2
+}
+```
+
+> 🔴 Giving a bad formed `_geo` that do not conform to the format causes the `update` payload to fail. A new `invalid_geo_field` error is given in the `update` object.
+
+##### `invalid_geo_field` error
+
+| field     | value                                                                                                                |
+|-----------|----------------------------------------------------------------------------------------------------------------------|
+| message   | The _geo field is invalid.                                                                                           |
+| errorCode | invalid_geo_field                                                                                                    |
+| errorType | invalid_request_error                                                                                                |
+| errorLink | *Link to the dedicated error page. This page should give further links to indicate the expected JSON and CSV format* |
+
+---
+
+### **As an end-user, I want to filter documents within a geo radius.**
+
+- Introduce a `_geoRadius({lat}, {lng}, {distance_in_meters})` built-in filter rule  to `filter` documents in a geo radius.shape.
+
+**`_geoRadius` built-in filter rule definition**
+
+- Name: `_geoRadius`
+- Signature: ({lat:float}:required, {lng:float}:required, {distance_in_meters:int}:required)
+- Not required
+- `distance_in_meters` only accepts positive value.
+
+>  The `_geo` field has to be set in `filterableAttributes` setting by the developer to activate geo filtering capabilities at search.
+
+#### GET Search `/indexes/{indexUid}/search`
+
+```
+...&filter="brand=Mercedes AND _geoRadius(48.862725, 2.287592, 2000)"`
+```
+
+#### POST Search `/indexes/{indexUid}/search`
+
+```
+{
+    "filter": ["brand = Ferrari", "_geoRadius(48.862725, 2.287592, 2000)"]
+}
+```
+
+> 🔴 Specifying parameters that do not conform to the `_geoRadius` signature causes the API to return an `invalid_filter` error. The error message should indicate how `_geoRadius` should be used. See `_geoRadius` built-in filter rule definition part.
```

Comment:
This means that if a user does not specify a distance (e.g. `_geoRadius(0.0, 0.0)`) they will get the `invalid_filter` error, correct?

Reply:
Yes! 

It could be very interesting to have dedicated errors for each of these built-in filtering/sorting rules in the future!